src/target/riscv/batch.c

```diff
@@ -89,17 +90,28 @@ bool riscv_batch_full(struct riscv_batch *batch)
 	return riscv_batch_available_scans(batch) == 0;
 }
 
-int riscv_batch_run(struct riscv_batch *batch, bool resets_delays,
-		size_t reset_delays_after)
+static bool riscv_batch_was_busy(const struct riscv_batch *batch, size_t scan_i)
```

Comment:
Detail: 

- You may consider renaming `scan_i` to `scan_idx`, `scan_index` or similar (also in the other functions).
- The function can be renamed to `riscv_batch_was_scan_busy()`, making it clearer that this refers to a particular scan, not the whole batch.

```suggestion
static bool riscv_batch_was_scan_busy(const struct riscv_batch *batch, size_t scan_index)
```

Reply:
Addressed.